### PR TITLE
[mds-*] Add promise linter rules, and clean up half of agency

### PR DESCRIPTION
**Pull request by avatarneil**
## PR Checklist

 - [x] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [x] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [x] Provider
- [x] Agency
- [x] Audit
- [x] Policy
- [x] Compliance

### CODE REVIEW

File: packages/mds-agency/api.ts [hunk 1]
```diff
@@ -163,28 +163,26 @@ function api(app: express.Express): express.Express {
...
-      },
-      (err: object) => /* istanbul ignore next */ {
-        log.error('failed to read service areas', err)
-        res.status(404).send({
-          result: 'not found'
-        })
-      }
-    )
+  app.get(pathsFor('/service_areas'), async (req: AgencyApiRequest, res: AgencyApiResponse) => {
+    const serviceAreas = await areas.readServiceAreas().catch((err: Error) => {
```
Comment (mdurling) [inline packages/mds-agency/api.ts]: I don't like mixing await with promises in this manner. In this case `serviceAreas` will probably be inferred as `ServiceArea[] | void` which makes typing of responses problematic. IMO, a better pattern is:
```
try {
  const serviceAreas = await areas.readServiceAreas()
  log.info('readServiceAreas (all)', serviceAreas.length)
  return res.status(200).send({ service_areas: serviceAreas })
} 
catch (err: Error) {
  log.error('failed to read service areas', err)
  return res.status(404).send({
    result: 'not found'
  })
}
```

Comment (avatarneil) [reply packages/mds-agency/api.ts]: @mdurling What do you think about the use of `.catch` with `await` in `/vehicles/:device_id`?

Comment (mdurling) [reply packages/mds-agency/api.ts]: I only pointed out one instance of the pattern. The others should be changed as well. Mixing `await` with `.catch` will lead to type confusion at some level.

Comment (avatarneil) [reply packages/mds-agency/api.ts]: The original instance of the pattern you pointed out made sense to resolve, however I don't see the reasoning with not using `.catch` after an `await`. There are some cases where you want to handle a rejection from promise `a` differently than promise `b`, as is demonstrated quite well by the `/vehicles/:device_id` example. Considering we don't have any sort of error code specification, and a function may only reject and not throw an error, I think handling it like this is the best we can do. Any type confusion that would arise can always be resolved by type-guarding, ```typescript
const res = await someAsyncFunction().catch(err => {
  log.warn(err)
  return undefined
})
if (!(res instanceOf ExpectedType)) {
  ...
}
rest of code that depends on res
```

If you can find a clean, concise way to handle it using just `try/catch` blocks with `async/await`, I'd love to see it.

**Comment by mdurling** [comment]
A function that rejects should still be caught in try/await/catch. In my experience await/.catch has always been replaceable with try/await/catch, but I defer to your judgement on this particular case.

**Pull request merged**